README.md

```diff
@@ -1,2 +1,21 @@
 # comp110-worksheet-C
 Base repository for COMP110 worksheet C
+
+
+a) The algo is describing a list comparison/duplication checking (if used in game development: possible collision avoidance--)
+
+b) The algo is moving over the list for each list item (If the algo does "n" things and each "thing is O(n) then the overall algo is O(n^2)
+
+c) 
+
+d) Because it is only checking the positions before the i (each iteration), not the whole list.
```

Comment:
✔️ though could be more specific about why this halves the execution time (i.e. because half the comparisons in the original algorithm were redundant)